extensions/shared/premium-blocks/index.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { uniq } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import domReady from '@wordpress/dom-ready';
+import { addFilter } from '@wordpress/hooks';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import withUpgradeBanner from './with-upgrade-banner';
+import { isUpgradeNudgeEnabled, isUpgradable } from '../plan-utils';
+import premiumBlockEdit from './edit';
+import renderPremiumIcon from './render-premium-icon.js';
+
+import './editor.scss';
+
+const jetpackPremiumBlock = ( settings, name ) => {
+	if ( isUpgradable( name ) ) {
+		// Populate block keywords.
+		settings.keywords = uniq( [ ...settings.keywords, 'premium', __( 'premium' ) ] );
+
+		// Extend BlockEdit function.
+		settings.edit = premiumBlockEdit( settings.edit );
+		settings.icon = renderPremiumIcon( settings.icon, name );
```

Comment:
Can the `name` parameter be removed? It doesn't look like `renderPremiumIcon()` expects it.

Reply:
Sure, I just forgot to remove it https://github.com/Automattic/jetpack/pull/16611/commits/6730cd0471e89a555978c4972560dd526eeda472